### PR TITLE
fix: remove query param from manifest entry of img

### PR DESCRIPTION
**Pull request by mchandak29**
Manifest entry for Images contain query param, eg: `/content/screens/conference-rooms/Gather/media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=750&format=webply&optimize=medium`

As a result, players download low resolution images.

This changes removes the query param in path for Images.

### CODE REVIEW

File: src/createManifest.js [hunk 1]
```diff
@@ -25,6 +25,11 @@ export default class ManifestGenerator {
   static trimImagesPath = (item, index, arr) => {
     const item1 = item.trim();
     arr[index] = item1[0] === '.' ? item1.substring(1, item1.length) : item1;
+    // remove query param from image path if present
+    const queryIndex = arr[index].indexOf('?');
```
Comment (div8063) [inline src/createManifest.js:29]: Can be simplified to 
```suggestion
    arr[index] = arr[index].split('?')[0];
```

File: src/createManifest.js [hunk 2]
```diff
@@ -94,15 +99,15 @@ export default class ManifestGenerator {
       resourceEntry.path = resourcesArr[i];
       // timestamp is optional value, only add if last-modified available
       const date = resp.headers.get('last-modified');
-      if (date) {
+      if (ManifestGenerator.isMedia(resourceSubPath)) {
```
Comment (div8063) [inline src/createManifest.js:102]: Trying to understand why the ordering is reversed here.
Can you please explain?

**Review by div8063 — CHANGES_REQUESTED** [review]

**Review by div8063 — APPROVED** [review]

**Comment by habansal** [comment]
@mchandak29 Where were these query params coming from?

**Comment by absarasw** [comment]
`Where were these query params coming from?`
@mchandak29 , I think we should understand the scenarios first. May be later on there can be a use-case where the parameters are required.

**Comment by mchandak29** [comment]
> Where were these query params coming from?

For every media file, Franklin adds these query params in markup to optimise for smaller resolution displays. The default width is used as "750". So, If viewport width is not defined (when you programmatically fetch the markup) - it returns the image downsampled for default width.

```
<picture>
    <source type="image/webp" srcset="[./media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=2000&#x26;format=webply&#x26;optimize=medium](https://main--cec-screens--hlxscreens.hlx.live/content/screens/conference-rooms/media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=2000&#x26;format=webply&#x26;optimize=medium)" media="(min-width: 600px)">
    <source type="image/webp" srcset="[./media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=750&#x26;format=webply&#x26;optimize=medium](https://main--cec-screens--hlxscreens.hlx.live/content/screens/conference-rooms/media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=750&#x26;format=webply&#x26;optimize=medium)">
    <source type="image/jpeg" srcset="[./media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=2000&#x26;format=jpeg&#x26;optimize=medium](https://main--cec-screens--hlxscreens.hlx.live/content/screens/conference-rooms/media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=2000&#x26;format=jpeg&#x26;optimize=medium)" media="(min-width: 600px)">
    <img loading="lazy" alt="" src="[./media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=750&#x26;format=jpeg&#x26;optimize=medium](https://main--cec-screens--hlxscreens.hlx.live/content/screens/conference-rooms/media_1b41e396687e973df6f4e84d3e545c2dc888ce1e0.jpeg?width=750&format=jpeg&optimize=medium)" width="1280" height="720">
</picture>
```

We however display these images on a different(mostly larger viewport width) sizes. By removing the query params, player fetches the image in it's highest quality.

> May be later on there can be a use-case where the parameters are required.

since these params are only downsampling images(to save network bandwidth) keeping the same aspect ratio. I don't see any use-case for it right now. wdyt?

cc: @habansal @absarasw

**Comment by habansal** [comment]
@mchandak29 Thanks for the explanation. Looks good!

**Pull request merged**